mlflow/pyfunc/scoring_server/client.py

```diff
@@ -11,6 +14,27 @@
 from mlflow.deployments import PredictionsResponse
 
 
+def dump_data(data):
+    def get_jsonable_input(name, data):
+        if isinstance(data, np.ndarray):
+            return data.tolist()
+        else:
+            raise MlflowException(f"Incompatible input type:{type(data)} for input {name}.")
+
+    if isinstance(data, pd.DataFrame):
+        post_data = {"dataframe_split": data.to_dict(orient="split")}
+    elif isinstance(data, dict):
+        post_data = {"inputs": {k: get_jsonable_input(k, v) for k, v in data}}
+    elif isinstance(data, np.ndarray):
+        post_data = {"inputs": data.tolist()}
+    else:
+        post_data = data
```

Comment:
Nit: Looks like we use similar logic in other places, e.g. https://github.com/mlflow/mlflow/blob/da9f8eef78ce1bf42c3a8aa8edf7c9f674d8538a/mlflow/sagemaker/__init__.py#L2689-L2692.

Is it possible to turn this into a reusable function?

Reply:
Yes!